packages/@aws-cdk/aws-efs/README.md

```diff
@@ -15,4 +15,67 @@
 ---
 <!--END STABILITY BANNER-->
 
+This construct library allows you to set up AWS Elastic File System (EFS).
+
+```ts
+import efs = require('@aws-cdk/aws-efs');
+
+const myVpc = new ec2.Vpc(this, 'VPC');
+const fileSystem = new efs.EfsFileSystem(this, 'MyEfsFileSystem', {
+  vpc: myVpc,
+  encrypted: true,
+  lifecyclePolicy: EfsLifecyclePolicyProperty.AFTER_14_DAYS,
+  performanceMode: EfsPerformanceMode.GENERAL_PURPOSE,
+  throughputMode: EfsThroughputMode.BURSTING
+});
+```
+
+### Connecting
+
+To control who can access the EFS, use the `.connections` attribute. EFS has
+a fixed default port, so you don't need to specify the port:
+
+```ts
+fileSystem.connections.allowDefaultPortFrom(instance);
+```
+### Mounting the file system using User Data
+
+In order to automatically mount this file system during instance launch, 
+following code can be used as reference:
+```
+const vpc = new ec2.Vpc(this, 'VPC');
+
+const fileSystem = new efs.EfsFileSystem(this, 'EfsFileSystem', {
+  vpc,
+  encrypted: true,
+  lifecyclePolicy: efs.EfsLifecyclePolicyProperty.AFTER_14_DAYS,
+  performanceMode: efs.EfsPerformanceMode.GENERAL_PURPOSE,
+  throughputMode: efs.EfsThroughputMode.BURSTING
+});
+
+const inst = new Instance(this, 'inst', {
+  instanceType: InstanceType.of(InstanceClass.T2, InstanceSize.LARGE),
+  machineImage: new AmazonLinuxImage({
+    generation: AmazonLinuxGeneration.AMAZON_LINUX_2
+  }),
+  vpc,
+  vpcSubnets: {
+    subnetType: SubnetType.PUBLIC,
+  }
+});
+
+fileSystem.connections.allowDefaultPortFrom(inst);
+
+inst.userData.addCommands("yum check-update -y",    // Ubuntu: apt-get -y update
```

Comment:
Do we need to worry about this at all? I don't see IAM permissions mentioned anywhere in your example:

https://docs.aws.amazon.com/efs/latest/ug/iam-access-control-nfs-efs.html

Reply:
> The default EFS file system policy grants full access to any NFS client that can connect to the file system.


In this example, EFS was not created using any profile permissions, the instance is mounting the file system and hence does require any specific IAM role assigned to it.

Reply:
Also, the IAM feature was [recently introduced](https://aws.amazon.com/blogs/aws/new-for-amazon-efs-iam-authorization-and-access-points/) to EFS. Since I started working on the PR before this feature's launch, I haven't added it here and will add this feature in subsequent PR.